### DISCUSSION TITLE
How to change the default port `51820` of Flannel Wireguard?

### DISCUSSION THREAD

**Discussion by LucienShui** [k3s-io/k3s#10784] [Q&A]
I have a cluster with the IP address xxx. Now, I have two more machines under the same public IP address yyy.

I started the server with `--node-external-ip=xxx --flannel-backend=wireguard-native --flannel-external-ip` according the doc: https://docs.k3s.io/networking/distributed-multicloud

Now, I want to add these two new nodes to the cluster using `--node-external-ip=yyy`.

However, I realized that k3s uses the fixed port 51820 for WireGuard connections. I can forward port 51820 for one machine to the IP address yyy, but I can't do it for both machines simultaneously.

Is there any way to configure Flannel WireGuard to use a port other than 51820?

**Comment by brandond** [comment]
No, and either way it is expected to be the same for all cluster members - so simply being able to change it would not solve the problem.

Node addresses are also expected to be unique, do not join multiple nodes to the cluster with the same internal or external IPs or hostnames.

**Reply by LucienShui** [reply]
Got it, thanks.

Now I have manually created a P2P WireGuard network to allow all nodes to communicate with each other using the WG IP.

| Hostname | IP | WG IP |
| --- | --- | --- |
| server-0 | x.x.x.x | 10.19.98.2 |
| server-1 | y.y.y.y | 10.19.98.3 |
| server-2 | z.z.z.z | 10.19.98.4 |
| node-0 | 192.168.1.2 | 10.19.98.5 |
| node-1 | 192.168.1.3 | 10.19.98.6 |
| node-2 | a.a.a.a | 10.19.98.7 |
| node-3 | b.b.b.b | 10.19.98.8 |

Where `x.x.x.x` stands for a public IP.

My question is, when I want to create a k3s cluster:

1. Should I enable Flannel?
2. What external IP should I use?
   + Should I simply use all WG IPs?
   + Or use the IPs or fallback to WG IPs?
